backend/src/schema/types/type/Badge.gql

```diff
@@ -10,4 +9,23 @@ type Badge {
   updatedAt: String
 
   rewarded: [User]! @relation(name: "REWARDED", direction: "OUT")
-}
\ No newline at end of file
+}
+
+enum BadgeStatus {
+  permanent
+  temporary
+}
+
+enum BadgeType {
+  role
+  crowdfunding
+}
+
+type Query {
+  Badge: [Badge]
+}
+
+type Mutation {
+  reward(badgeKey: ID!, userId: ID!): User
+  unreward(badgeKey: ID!, userId: ID!): User
+}
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-07-11T14:44:26Z" title="Thursday, July 11th 2019, 4:44:26 pm +02:00">Jul 11, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -10,4 +9,23 @@ type Badge {
   updatedAt: String
 
   rewarded: [User]! @relation(name: "REWARDED", direction: "OUT")
-}
\ No newline at end of file
+}
+
+enum BadgeStatus {
+  permanent
+  temporary
+}
+
+enum BadgeType {
+  role
+  crowdfunding
+}
+
+type Query {
+  Badge: [Badge]
+}
+
+type Mutation {
+  reward(badgeKey: ID!, userId: ID!): User
+  unreward(badgeKey: ID!, userId: ID!): User
+}
```


I am a fan of this refactoring